File: Stratego Game/Week 1 iLab/sprite.cpp
#include "sprite.h"

The_Sprite::The_Sprite()
{
	texture = NULL;
	position = D3DXVECTOR3(0, 0, 0);
	scalex = 0;
	scaley = 0;
	rotation = 0;
	rect.bottom = 0;
	rect.top = 0;
	rect.left = 0;
	rect.right = 0;
}

The_Sprite::The_Sprite(const The_Sprite & s)
{
	texture = s.texture;
	position = s.position;
	scalex = s.scalex;
	scaley = s.scaley;
	rotation = s.rotation;
	rect.bottom = s.rect.bottom;
	rect.top = s.rect.top;
	rect.left = s.rect.left;
	rect.right = s.rect.right;
}

The_Sprite::~The_Sprite()
{
}

void The_Sprite::setTexture(IDirect3DTexture9* t)
{
	texture = t;
}

IDirect3DTexture9* The_Sprite::getTexture()
{
	return texture;
}

void The_Sprite::setPosition(D3DXVECTOR3 p)
{
	position = p;
}

D3DXVECTOR3 The_Sprite::getPosition()
{
	return position;
}

void The_Sprite::setScalex(float s)
{
	scalex = s;
}

float The_Sprite::getScalex()
{
	return scalex;
}

void The_Sprite::setScaley(float s)
{
	scaley = s;
}

float The_Sprite::getScaley()
{
	return scaley;
}

void The_Sprite::setRotation(float r)
{
	rotation = r;
}

float The_Sprite::getRotation()
{
	return rotation;
}

void The_Sprite::setRect(RECT r)
{
	rect = r;
}

RECT The_Sprite::getRect()
{
	return rect;
}

File: Stratego Game/Week 1 iLab/Sprite.java
package StrategoJava;

import java.awt.Graphics;
import java.awt.Image;

public class Sprite {

	//Texture texture;
	protected Image texture;
	protected Vector2D position;

	protected float scalex;
	protected float scaley;
	protected float rotation;
	protected Rect rect;
	
	public Sprite()
	{
		texture = null;
		position = new Vector2D();
		scalex = 0;
		scaley = 0;
		rotation = 0;
	}
	
	public void init(Image i, Vector2D a_v, float a_s_x, float a_s_y, float a_r/*, RECT recta */)
	{
		texture = i;
		position = a_v;
		scalex = a_s_x;
		scaley = a_s_y;
		rotation = a_r;
	}
	
	public void setTexture(Image i) {texture = i;}
	public Image getTexture() {return texture;}
	public void setPosition(Vector2D a_v) {position = a_v;}
	public Vector2D getPosition() {return position;}
	public void setScaleX(float a_s_x) {scalex = a_s_x;}
	public void setScaleY(float a_s_y) {scaley = a_s_y;}
	public void setRotation(float a_r) {rotation = a_r;}
	public float getScaleX() {return scalex;}
	public float getScaleY() {return scaley;}
	public float getRotation() {return rotation;}
	public void setRect(Rect r) {rect = r;}
	public Rect getRect() {return rect;}
	
	public void draw(Graphics g)
	{
		g.drawImage(texture, position.getX(), position.getY(), null);
	}
};
